### DISCUSSION TITLE
WaveSurfer is not working

### DISCUSSION THREAD

**Discussion by webdevratnesh** [katspaugh/wavesurfer.js#3161] [General]
Hi memebers,

I have copied code from :- https://wavesurfer-js.org/ and renamed the audio file name accordingly but wavesurfer is not working. what could be the possible reason.

Here's the SS of code and webpage:- https://prnt.sc/d--xHpw0aDnE

Please tell me what I am missing.

I have used below code:-

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div id="waveform">
        <!-- the waveform will be rendered here -->
      </div>
      
      <script type="module">
      import WaveSurfer from 'https://unpkg.com/wavesurfer.js@7/dist/wavesurfer.esm.js'
      
      const wavesurfer = WaveSurfer.create({
        container: '#waveform',
        waveColor: '#4F4A85',
        progressColor: '#383351',
        url: 'https://www.jekkna.com/wp-content/uploads/2023/08/Home-Audio-JEKKNA.mp3',
      })
      
      wavesurfer.on('interaction', () => {
        wavesurfer.play()
      })
      </script>
</body>
</html>
```

**Comment by katspaugh** [comment]
It won't work when opened as an HTML file via `C:/Users/...`. You need to run a local web server to serve your HTML.
Besides, loading just any audio from the Internet won't work either because of [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). You need to either host the audio yourself or the server must support CORS.

**Reply by samp101** [reply]
I am having cors issues when trying to load the waveSurfer.js waves. instead of enabling all cors is it possible to enable it that my server will allow wavesurfers IP address. If yes then what is the IP address?

**Reply by katspaugh** [reply]
It's the domain (not IP) from which you serve your HTML.

**Comment by webdevratnesh** [comment]
Thanks for your reply. My issue got resolved :)
